### PR TITLE
Story/959/5023 - Data tabulator via configuration

### PR DESCRIPTION
**Pull request by ja6a-regnosys**
Allow DSL users to configure tabulators to be generated for specified types. E.g.:

```
model:
  name: My test model
dependencies:
generators:
  namespaces:
  - model1.*
  tabulators:
    annotations:
    types:
    - model1.Foo
    - model1.Bar
```
## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend [hunk 1]
```diff
@@ -218,6 +255,12 @@ class TabulatorGenerator {
 		}
 	}
 	
+	private def boolean isTypeTabulatable(Data type) {
+		val types = rosettaConfiguration.generators.tabulators.types
+		val fqn = String.format("%s.%s", type.model.name, type.name)
```
Comment (ja6a-regnosys) [inline rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend:260]: i will push an update to this now ... ;-)

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend [hunk 2]
```diff
@@ -136,7 +133,31 @@ class TabulatorGenerator {
 		override getRule(Attribute attr) {
 			Optional.empty
 		}
+	}
+	@org.eclipse.xtend.lib.annotations.Data
+	private static class DataTabulatorContext implements TabulatorContext {
+		extension JavaTypeTranslator typeTranslator
+		Data data
```
Comment (SimonCockx) [inline rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend:140]: this is never used

Comment (ja6a-regnosys) [reply rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend:140]: This is necessary!

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend (hunk 1 above)
Comment (SimonCockx) [reply rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend]: I think this is alright for now. In our POC I've created a utility for this, but that's not production code yet...

File: rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend (hunk 2 above)
Comment (SimonCockx) [reply rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend:140]: the `data` field?

Comment (ja6a-regnosys) [reply rosetta-lang/src/main/java/com/regnosys/rosetta/generator/java/reports/TabulatorGenerator.xtend]: correction - no it isn't - now removed.

**Review by davidalk — APPROVED** [review]

**Review by SimonCockx — APPROVED** [review]

**Pull request merged**